From 56c2ac081ada4f09dae665d3c9f86e8dbd725688 Mon Sep 17 00:00:00 2001
From: "R. Andrew Ohana" <andrew.ohana@gmail.com>
Date: Sun, 24 Mar 2013 14:34:10 -0700
Subject: [PATCH] (FIXUP) don't reference devel/sage in doctests

---

diff --git a/__SAGE_SRC__/doc/common/builder.py b/__SAGE_SRC__/doc/common/builder.py
index 87cd074..ce051bf 100644
--- a/__SAGE_SRC__/doc/common/builder.py
+++ b/__SAGE_SRC__/doc/common/builder.py
@@ -139,7 +139,7 @@ class DocBuilder(object):
             sage: import os, sys; sys.path.append(os.environ['SAGE_DOC']+'/common/'); import builder
             sage: b = builder.DocBuilder('tutorial')
             sage: b._output_dir('html')
-            '.../devel/sage/doc/output/html/en/tutorial'
+            '.../doc/output/html/en/tutorial'
         """
         d = os.path.join(SAGE_DOC, "output", type, self.lang, self.name)
         mkdir(d)
@@ -156,7 +156,7 @@ class DocBuilder(object):
             sage: import os, sys; sys.path.append(os.environ['SAGE_DOC']+'/common/'); import builder
             sage: b = builder.DocBuilder('tutorial')
             sage: b._doctrees_dir()
-            '.../devel/sage/doc/output/doctrees/en/tutorial'
+            '.../doc/output/doctrees/en/tutorial'
         """
         d = os.path.join(SAGE_DOC, "output", 'doctrees', self.lang, self.name)
         mkdir(d)
@@ -442,7 +442,7 @@ class ReferenceBuilder(AllBuilder):
             sage: import os, sys; sys.path.append(os.environ['SAGE_DOC']+'/common/'); import builder
             sage: b = builder.ReferenceBuilder('reference')
             sage: b._output_dir('html')
-            '.../devel/sage/doc/output/html/en/reference'
+            '.../doc/output/html/en/reference'
         """
         d = os.path.join(SAGE_DOC, "output", type, lang, self.name)
         mkdir(d)
@@ -907,7 +907,7 @@ class ReferenceSubBuilder(DocBuilder):
             sage: import os, sys; sys.path.append(os.environ['SAGE_DOC']+'/common/'); import builder
             sage: import builder
             sage: builder.ReferenceSubBuilder("reference").auto_rest_filename("sage.combinat.partition")
-            '.../devel/sage/doc/en/reference/sage/combinat/partition.rst'
+            '.../doc/en/reference/sage/combinat/partition.rst'
         """
         return self.dir + os.path.sep + module_name.replace('.',os.path.sep) + '.rst'
 
diff --git a/__SAGE_SRC__/sage/misc/cython.py b/__SAGE_SRC__/sage/misc/cython.py
index 36ff57e..5faeef4 100644
--- a/__SAGE_SRC__/sage/misc/cython.py
+++ b/__SAGE_SRC__/sage/misc/cython.py
@@ -212,9 +212,9 @@ def pyx_preparse(s):
         '.../include',
         '.../include/python2.7',
         '.../lib/python/site-packages/numpy/core/include',
-        '.../devel/sage/sage/ext',
-        '.../devel/sage',
-        '.../devel/sage/sage/gsl'],
+        '.../sage/ext',
+        '...',
+        '.../sage/gsl'],
         'c',
         [], ['-w', '-O2'])
         sage: s, libs, inc, lang, f, args = pyx_preparse("# clang c++\n #clib foo\n # cinclude bar\n")
@@ -240,9 +240,9 @@ def pyx_preparse(s):
         '.../include',
         '.../include/python2.7',
         '.../lib/python/site-packages/numpy/core/include',
-        '.../devel/sage/sage/ext',
-        '.../devel/sage',
-        '.../devel/sage/sage/gsl']
+        '.../sage/ext',
+        '...',
+        '.../sage/gsl']
 
         sage: s, libs, inc, lang, f, args = pyx_preparse("# cargs -O3 -ggdb\n")
         sage: args
-- 
1.8.1.5

