From 7f26a4f14df3adef0d1499f5fb9542b91aef6fc2 Mon Sep 17 00:00:00 2001
From: "R. Andrew Ohana" <andrew.ohana@gmail.com>
Date: Sat, 23 Mar 2013 14:12:45 -0700
Subject: [PATCH] (FIXUP) Makefile: put logs in logs directory

---

diff --git a/Makefile b/Makefile
index 7dfa378..58c4f9f 100644
--- a/Makefile
+++ b/Makefile
@@ -17,7 +17,7 @@ build:
 	cd build && \
 	"../$(PIPE)" \
 		"env SAGE_PARALLEL_SPKG_BUILD='$(SAGE_PARALLEL_SPKG_BUILD)' ./install all 2>&1" \
-		"tee -a ../install.log"
+		"tee -a ../logs/install.log"
 	./sage -b
 
 # ssl: build Sage, and also install pyOpenSSL. This is necessary for
@@ -49,16 +49,16 @@ start: build
 doc: doc-html
 
 doc-html: build
-	$(PIPE) "./sage --docbuild --no-pdf-links all html $(SAGE_DOCBUILD_OPTS) 2>&1" "tee -a dochtml.log"
+	$(PIPE) "./sage --docbuild --no-pdf-links all html $(SAGE_DOCBUILD_OPTS) 2>&1" "tee -a logs/dochtml.log"
 
 doc-html-mathjax: build
-	$(PIPE) "./sage --docbuild --no-pdf-links all html -j $(SAGE_DOCBUILD_OPTS) 2>&1" "tee -a dochtml.log"
+	$(PIPE) "./sage --docbuild --no-pdf-links all html -j $(SAGE_DOCBUILD_OPTS) 2>&1" "tee -a logs/dochtml.log"
 
 # Keep target 'doc-html-jsmath' for backwards compatibility.
 doc-html-jsmath: doc-html-mathjax
 
 doc-pdf: build
-	$(PIPE) "./sage --docbuild all pdf $(SAGE_DOCBUILD_OPTS) 2>&1" "tee -a docpdf.log"
+	$(PIPE) "./sage --docbuild all pdf $(SAGE_DOCBUILD_OPTS) 2>&1" "tee -a logs/docpdf.log"
 
 doc-clean:
 	@echo "Deleting devel/sage/doc/output..."
@@ -107,37 +107,37 @@ TESTDIRS = src/doc/common src/doc/[a-z][a-z] src/sage
 
 test: all # i.e. build and doc
 	$(TESTPRELIMS)
-	$(PIPE) "./sage -t --sagenb $(TESTDIRS) 2>&1" "tee -a test.log"
+	$(PIPE) "./sage -t --sagenb $(TESTDIRS) 2>&1" "tee -a logs/test.log"
 
 check: test
 
 testall: all # i.e. build and doc
 	$(TESTPRELIMS)
-	$(PIPE) "./sage -t --sagenb --optional $(TESTDIRS) 2>&1" "tee -a testall.log"
+	$(PIPE) "./sage -t --sagenb --optional $(TESTDIRS) 2>&1" "tee -a logs/testall.log"
 
 testlong: all # i.e. build and doc
 	$(TESTPRELIMS)
-	$(PIPE) "./sage -t --sagenb --long $(TESTDIRS) 2>&1" "tee -a testlong.log"
+	$(PIPE) "./sage -t --sagenb --long $(TESTDIRS) 2>&1" "tee -a logs/testlong.log"
 
 testalllong: all # i.e. build and doc
 	$(TESTPRELIMS)
-	$(PIPE) "./sage -t --sagenb --optional --long $(TESTDIRS) 2>&1" "tee -a testalllong.log"
+	$(PIPE) "./sage -t --sagenb --optional --long $(TESTDIRS) 2>&1" "tee -a logs/testalllong.log"
 
 ptest: all # i.e. build and doc
 	$(TESTPRELIMS)
-	$(PIPE) "./sage -tp --sagenb $(TESTDIRS) 2>&1" "tee -a ptest.log"
+	$(PIPE) "./sage -tp --sagenb $(TESTDIRS) 2>&1" "tee -a logs/ptest.log"
 
 ptestall: all # i.e. build and doc
 	$(TESTPRELIMS)
-	$(PIPE) "./sage -tp --sagenb --optional $(TESTDIRS) 2>&1" "tee -a ptestall.log"
+	$(PIPE) "./sage -tp --sagenb --optional $(TESTDIRS) 2>&1" "tee -a logs/ptestall.log"
 
 ptestlong: all # i.e. build and doc
 	$(TESTPRELIMS)
-	$(PIPE) "./sage -tp --sagenb --long $(TESTDIRS) 2>&1" "tee -a ptestlong.log"
+	$(PIPE) "./sage -tp --sagenb --long $(TESTDIRS) 2>&1" "tee -a logs/ptestlong.log"
 
 ptestalllong: all # i.e. build and doc
 	$(TESTPRELIMS)
-	$(PIPE) "./sage -tp --sagenb --optional --long $(TESTDIRS) 2>&1" "tee -a ptestalllong.log"
+	$(PIPE) "./sage -tp --sagenb --optional --long $(TESTDIRS) 2>&1" "tee -a logs/ptestalllong.log"
 
 
 testoptional: testall # just an alias
-- 
1.8.1.5

