From 4109caa838ab0e6cef60a1ca2391216a812c374a Mon Sep 17 00:00:00 2001
From: "R. Andrew Ohana" <andrew.ohana@gmail.com>
Date: Sun, 17 Mar 2013 01:01:23 -0700
Subject: [PATCH] (FIXUP) update sage-starts

---

diff --git a/__SAGE_SCRIPTS_DIR__/sage-starts b/__SAGE_SCRIPTS_DIR__/sage-starts
index d9f034a..6b86750 100755
--- a/__SAGE_SCRIPTS_DIR__/sage-starts
+++ b/__SAGE_SCRIPTS_DIR__/sage-starts
@@ -14,7 +14,7 @@ echo "[`date +'%Y-%m-%d %H:%M:%S'`] `cat VERSION.txt 2>/dev/null`" | tee -a star
 # sysadmin.  We use --nodotsage so we don't force a .sage directory
 # in the sysadmin's HOME directory.
 cmd='sage.all._write_started_file(); print "Yes, Sage starts."'
-spkg/pipestatus "./sage --nodotsage -c '$cmd' 2>&1" "tee -a start.log"
+__SAGE_BUILD__/pipestatus "./sage --nodotsage -c '$cmd' 2>&1" "tee -a start.log"
 
 if [ $? -ne 0 ]; then
     echo >&2 "Sage failed to start up."
-- 
1.8.1.5

